packages/react/context-menu/src/ContextMenu.tsx

```diff
@@ -20,12 +22,42 @@ type ContextMenuContextValue = {
   onOpenChange(open: boolean): void;
 };
 
+const SubmenuContext = React.createContext<boolean | undefined>(undefined);
+const InsideContentContext = React.createContext(false);
 const [ContextMenuProvider, useContextMenuContext] = createContext<ContextMenuContextValue>(
   CONTEXT_MENU_NAME
 );
 
-const ContextMenu: React.FC<{ onOpenChange?(open: boolean): void }> = (props) => {
-  const { children, onOpenChange } = props;
+type ContextMenuOwnProps = React.ComponentProps<typeof ContextMenuImpl>;
+
+const ContextMenu: React.FC<ContextMenuOwnProps> = (props) => {
+  const parentSubmenuContext = React.useContext(SubmenuContext);
+  const isInsideContent = React.useContext(InsideContentContext);
+  const isRootMenu = parentSubmenuContext === undefined;
+
+  return (
+    <SubmenuContext.Provider value={!isRootMenu && isInsideContent}>
```

Comment:
Actually wait... do we even need the `isRootMenu` stuff in `ContextMenu` either now? If we're always checking `isInsideContent` then it's inherently not a root menu if it's inside content right? I feel like I've forgotten something here 🙈😅